clang-tools-extra/clang-tidy/daedalean/OperatorOverloadingCheck.cpp

```diff
@@ -0,0 +1,213 @@
+//===--- OperatorOverloadingCheck.cpp - clang-tidy ------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "OperatorOverloadingCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace daedalean {
+
+namespace  {
+AST_MATCHER(FunctionDecl, isOverloadedOperator) {
+  if (const auto *CXXMethodNode = dyn_cast<CXXMethodDecl>(&Node)) {
+    if (CXXMethodNode->getParent()->isLambda())
+      return false;
+  }
```

Comment:
why does lambda have to be excluded manually?

Reply:
Because we no not want to check operators generated by compiler for lambda type

Reply:
Okay, but doesn't this apply to *all* operators generated by the compiler? I thought I saw something like `isImplicit()` somewhere, would it be more fitting to check for that?